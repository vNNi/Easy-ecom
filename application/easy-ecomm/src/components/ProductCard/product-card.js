import { useState } from 'react';

import { HeartOutlined, HeartFilled } from '@ant-design/icons';

import { currencyFormatter } from '../../utils';
import {
  Container,
  IconContainer,
  ImageContainer,
  InfosContainer,
  Title,
  InStockLabel,
  StockValue,
  Price,
} from './product-card.styles';

const ProductCard = ({
  id,
  title,
  price,
  image = 'https://ninajohansson.se/wp-content/themes/koji/assets/images/default-fallback-image.png',
  stock_count: stockCount,
  onClick = () => ({}),
  onFavorite = () => ({}),
  isFavorited,
}) => {
  const [favorited, setFavorited] = useState(isFavorited);
  const handleFavoriteClick = () => {
    onFavorite({ id, title, price, image, stock_count: stockCount });
    setFavorited(true);
  };

  const icon = favorited ? (
    <HeartFilled style={{ color: 'red' }} onClick={handleFavoriteClick} />
  ) : (
    <HeartOutlined onClick={handleFavoriteClick} />
  );
  return (
    <Container onClick={e => onClick(e, id)}>
      <IconContainer>{icon}</IconContainer>
      <ImageContainer>
        <img alt="" src={image} width={120} height={100} />
      </ImageContainer>
      <InfosContainer>
        <Title>{title}</Title>
        <div>
          <InStockLabel>Em estoque: </InStockLabel>
          <StockValue>{stockCount || 0}</StockValue>
        </div>
        <Price>{currencyFormatter({ value: Number(price) })}</Price>
      </InfosContainer>
    </Container>
  );
};

export default ProductCard;
