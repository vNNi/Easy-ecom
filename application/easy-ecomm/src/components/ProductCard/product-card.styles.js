import styled from 'styled-components';

export const Container = styled.h1`
  color: ${({ theme }) => theme.colors.primary};
`;
