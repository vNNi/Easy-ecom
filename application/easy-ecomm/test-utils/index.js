import React from 'react';
import { render } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';

import '@testing-library/jest-dom/extend-expect'; // jest-dom custom assertions
import 'jest-styled-components';

import Theme from '../src/styles/theme';

const AllTheProviders = ({ children }) => {
  return <ThemeProvider theme={Theme}>{children}</ThemeProvider>;
};

export const renderWithTheme = (ui, options) =>
  render(ui, { wrapper: AllTheProviders, ...options });

// re-export everything
export * from '@testing-library/react';
