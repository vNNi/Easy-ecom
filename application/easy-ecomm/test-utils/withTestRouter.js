import React from 'react';
import { RouterContext } from 'next/dist/next-server/lib/router-context';

export default function withTestRouter(tree, router = {}) {
  const {
    route = '',
    pathname = '',
    pagename = '',
    pagetype = '',
    query = {},
    asPath = '',
    push = async () => true,
    replace = async () => true,
    reload = () => null,
    back = () => null,
    prefetch = async () => undefined,
    beforePopState = () => null,
    isFallback = false,
    events = {
      emit: () => null,
      off: () => null,
      on: () => null,
    },
  } = router;

  return (
    <RouterContext.Provider
      value={{
        asPath,
        back,
        beforePopState,
        events,
        isFallback,
        pagename,
        pagetype,
        pathname,
        prefetch,
        push,
        query,
        reload,
        replace,
        route,
      }}
    >
      {tree}
    </RouterContext.Provider>
  );
}
