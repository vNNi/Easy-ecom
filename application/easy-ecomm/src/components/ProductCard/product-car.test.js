import { renderWithTheme } from '../../../test-utils';

import ProductCard from './product-card';

describe('[Components] - product card', () => {
  test('should expect', () => {
    const comp = renderWithTheme(<ProductCard />);
    expect(comp).toMatchSnapshot();
    expect(1).toBe(1);
  });
});
