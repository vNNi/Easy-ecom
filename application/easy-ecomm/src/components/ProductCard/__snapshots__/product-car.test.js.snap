// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Components] - product card should expect 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": .c0 {
  color: #0077b6;
}

<body>
    <div>
      <h1
        class="c0"
      >
        OI
      </h1>
    </div>
  </body>,
  "container": <div>
    <h1
      class="product-cardstyles__Container-sc-19h0a0q-0 jFDMfJ"
    >
      OI
    </h1>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
